HW_6/task_2.js

```diff
@@ -0,0 +1,263 @@
+            if(department.name === departmenName){
+                dep = obj.name;
+                return;                
+            }
+        });
+    });
+    
+    return dep
+}
+
+console.log(getEnterpriseName(1));
+console.log(getEnterpriseName1("Отдел маркетинга"))
+
+// 3. Написать функцию, которая будет добавлять предприятие. В качестве аргумента принимает название предприятия
+
+// Пример:
+// addEnterprise("Название нового предприятия")
+
+function addEnterprise(nameEnterprise){
+    let newObj = {name: nameEnterprise, id: enterprises.length +1, departments: []};
+    enterprises.push(newObj)   
+    return enterprises
+}
+
+console.log(addEnterprise('Предприятие Тест'))
+
+// 4. Написать функцию, которая будет добавлять отдел в предприятие. В качестве аргумента принимает id предприятия, в которое будет добавлен отдел и название отдела.
+
+// Пример:
+// addDepartment(1, "Название нового отдела")
+
+function addDepartmentToEnterprise(id, nameDepartment){
+    enterprises.forEach(obj => {        
+            if(obj.id === id){
+                let newDep = {name: nameDepartment, id: obj.departments.length-1, employees_count: 0};
+                obj.departments.push(newDep);                             
+            }
+        });
+    return enterprises
+};       
+
+
+console.log(addDepartmentToEnterprise(4, 'Автотестеры'))
+
+// 5. Написать функцию для редактирования названия предприятия. Принимает в качестве аргумента id предприятия и новое имя предприятия.
+
+// Пример:
+// editEnterprise(1, "Новое название предприятия")
+
+function editEnterprise(id, newFirmName){
+    enterprises.forEach(obj =>{
+        if(obj.id === id){
+            obj.name = newFirmName;
+        }
+    });
+    return enterprises
+};
+
+console.log(editEnterprise(4, "Предприятие 4"))
+
+
+// 6. Написать функцию для редактирования названия отдела. Принимает в качестве аргумента id отдела и новое имя отдела.
+
+// Пример:
+// editDepartment(7, "Новое название отдела")
+
+function editDepartment(id, newDepartment){
+    enterprises.forEach(obj =>{
+        if(obj.id === id){
+            obj.departments.name = newDepartment;             
+        };
+    });
+    return enterprises
+};
+
+console.log(editDepartment(4, "AQA"))
+
+// 7. Написать функцию для удаления предприятия. В качестве аргумента принимает id предприятия.
+
+// Пример:
+// deleteEnterprise(1)
+
+function deleteEnterprise(id){
+    enterprises.forEach(obj =>{
+        if(obj.id === id){
+            const index = enterprises.indexOf(obj)
+            enterprises.splice(index, 1)
+        }
+    });
+    return enterprises
+};
+
+console.log(deleteEnterprise(4))
+
+
+// 8. Написать функцию для удаления отдела. В качестве аргумента принимает id отдела. Удалить отдел можно только, если в нем нет сотрудников.
+
+// Пример:
+// deleteDepartment(3)
+
+function deleteDeparment(idOfDep) {
+    enterprises.forEach(obj => {
+        obj.departments.forEach(department => {

```

Comment:
Try to name arguments or variable to be as meaningful as possible
```suggestion
    enterprises.forEach(enterprise => {
       enterprise.departments.forEach(department => {
```